src/runtime/virtcontainers/clh.go

```diff
@@ -854,7 +853,17 @@ func clhDriveIndexToID(i int) string {
 // assumption convert a clh PciDeviceInfo into a PCI path
 func clhPciInfoToPath(pciInfo chclient.PciDeviceInfo) (types.PciPath, error) {
 	tokens := strings.Split(pciInfo.Bdf, ":")
-	if len(tokens) != 3 || tokens[0] != "0000" || tokens[1] != "00" {
+	if len(tokens) != 3 || tokens[1] != "00" {
+		return types.PciPath{}, fmt.Errorf("Unexpected PCI address %q from clh hotplug", pciInfo.Bdf)
+	}
+
+	// Support up to 10 PCI segments.
+	pciSegment, err := regexp.Compile(`^000[0-9]$`)
+	if err != nil {
+		return types.PciPath{}, fmt.Errorf("Internal error: cannot compile PCI segment regex")
+	}
+
+	if !pciSegment.MatchString(tokens[0]) {
 		return types.PciPath{}, fmt.Errorf("Unexpected PCI address %q from clh hotplug", pciInfo.Bdf)
```

Comment:
minor, please ignore if it slows progress: can we distinguish the error message from above error case? makes it easier to debug

Reply:
> minor, please ignore if it slows progress: can we distinguish the error message from above error case? makes it easier to debug

Sorry, I intentionally kept the original error messages from upstream - e.g., see:
https://github.com/kata-containers/kata-containers/blob/03554c799a96447460ce828b6293596ea61ba9ee/src/runtime/virtcontainers/clh.go#L817

I didn't want to disturb the Upstream universe more than I had to.

Reply:
ACK thanks for the clarification